cmd/flakechecker/main.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	_ "embed"
+	"flag"
+	"fmt"
+	"github.com/carlmjohnson/versioninfo"
+	junit "github.com/joshdk/go-junit"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/stackrox/junit2jira/pkg/testcase"
+	"os"
+)
+
+const totalRunsLimit = 30
+
+type flakeCheckerParams struct {
+	junitReportsDir string
+	configFile      string
+
+	jobName      string
+	orchestrator string
+
+	dryRun bool
+}
+
+func main() {
+	var debug bool
+	var err error
+
+	p := flakeCheckerParams{}
+	flag.StringVar(&p.junitReportsDir, "junit-reports-dir", os.Getenv("ARTIFACT_DIR"), "Dir that contains jUnit reports XML files")
+	flag.StringVar(&p.configFile, "config-file", "", "Config file with defined failure ratios")
+
+	flag.StringVar(&p.jobName, "job-name", "", "Name of CI job.")
+	flag.StringVar(&p.orchestrator, "orchestrator", "", "orchestrator name (such as GKE or OpenShift), if any.")
+
+	flag.BoolVar(&p.dryRun, "dry-run", false, "When set to true issues will NOT be created.")
+	flag.BoolVar(&debug, "debug", false, "Enable debug log level")
+	versioninfo.AddFlag(flag.CommandLine)
+	flag.Parse()
+
+	if debug {
+		log.SetLevel(log.DebugLevel)
+	}
+
+	err = p.run()
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+type recentFlakyTestsRecord struct {
+	JobName      string
+	FilteredName string
+	Classname    string
+	TotalAll     int
+	FailRatio    int
+}
+
+func (p *flakeCheckerParams) checkFailedTests(bqClient biqQueryClient, failedTests []testcase.TestCase, flakeConfigs []*flakeCheckerRecord) error {
+	for _, failedTest := range failedTests {
+		found := false
+		log.Infof("Checking failed test: %q / %q / %q", p.jobName, failedTest.Name, failedTest.Classname)
+		for _, flakeConfig := range flakeConfigs {
+			match, err := flakeConfig.matchJobName(p.jobName)
+			if err != nil {
+				return err
+			}
+
+			if !match {
+				continue
+			}
+
+			match, err = flakeConfig.matchTestName(failedTest.Name)
+			if err != nil {
+				return err
+			}
+
+			if !match {
+				continue
+			}
+
+			match, err = flakeConfig.matchClassname(failedTest.Classname)
+			if err != nil {
+				return err
+			}
+
+			if !match {
+				continue
+			}
+
+			found = true
+			log.Infof("Match found: %q / %q / %q", flakeConfig.MatchJobName, flakeConfig.TestNameRegex, flakeConfig.Classname)
+			totalRuns, failRatio, err := bqClient.GetRatioForTest(flakeConfig, failedTest.Name)
+			if err != nil {
+				return errors.New(fmt.Sprintf("Get ratio for test failed: %s", err))
```

Comment:
Why not errors.Wrap like in other places?
```suggestion
				return fmt.Errorf("Get ratio for test failed: %w", err)
```

Reply:
Also in other places we don't need double wrappring `errors.New(fmt.Sprintf` `fmt.Errorf` will be enough. Also errors should start from non capital letter